spec/models/tenant_spec.rb

```diff
@@ -586,6 +586,73 @@
     end
   end
 
+  describe ".used_quotas" do
+    let(:tenant) { FactoryGirl.create(:tenant, :parent => default_tenant) }
+    let(:vm1) do
+      FactoryGirl.create(
+        :vm_vmware,
+        :tenant_id => tenant.id,
+        :host_id   => 1,
+        :hardware  => FactoryGirl.create(
+          :hardware,
+          :memory_cpu   => 1024,
+          :logical_cpus => 1,
+          :disks        => [FactoryGirl.create(:disk, :size => 12_345_678, :size_on_disk => 12_345)]
+        )
+      )
+    end
+
+    let(:vm2) do
+      FactoryGirl.create(
+        :vm_vmware,
+        :tenant_id => tenant.id,
+        :host_id   => 1,
+        :hardware  => FactoryGirl.create(
+          :hardware,
+          :memory_cpu   => 1024,
+          :logical_cpus => 1,
+          :disks        => [FactoryGirl.create(:disk, :size => 12_345_678, :size_on_disk => 12_345)]
+        )
+      )
+    end
+
+    before do
+      tenant.set_quotas(
+        :vms_allocated       => {:value => 20},
+        :mem_allocated       => {:value => 4096},
+        :cpu_allocated       => {:value => 4},
+        :storage_allocated   => {:value => 123_456_789},
+        :templates_allocated => {:value => 10}
+      )
+    end
+
+    it "can get the used quota values" do
+      vm1.stub(:state => "on")
+      vm2.template = true
```

Comment:
Template is a separate model, I would expect to see something like `FactoryGirl.create(:miq_template)` here.  Since you are only counting the number of templates you can probably get away without having to setup the hardware.